### PR TITLE
[Merged by Bors] - chore: bump to nightly-2023-05-31

### PR DESCRIPTION
**Pull request by kim-em**
---

- [x] depends on: https://github.com/leanprover/std4/pull/150

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover/std4#150
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover/std4#150~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

File: Mathlib/CategoryTheory/Preadditive/HomOrthogonal.lean [hunk 1]
```diff
@@ -93,7 +93,12 @@ noncomputable def matrixDecomposition (o : HomOrthogonal s) {α β : Type} [Fint
     biproduct.matrix fun j k =>
       if h : f j = g k then z (f j) ⟨k, by simp [h]⟩ ⟨j, by simp⟩ ≫ eqToHom (by simp [h]) else 0
   left_inv z := by
-    ext (j k)
+    -- Porting note: `ext j k` used to achieve the next four steps,
```
Comment (digama0) [inline Mathlib/CategoryTheory/Preadditive/HomOrthogonal.lean:96]: This is not true, as I mentioned [elsewhere](https://github.com/leanprover-community/mathlib4/pull/4340#discussion_r1210974051); lean 3 applied `hom_ext'` first and the proof below does not reflect the original lean 3 proof.

**Comment by digama0** [comment]
This PR conflicts with #4340, they should be combined.

**Pull request closed without merging**